### ISSUE TITLE
Metadata of Charters

### ISSUE THREAD

**Issue by StephanMa** [icaruseu/mom-ca#559]
From an IT-Point-of-view the current solution with storing an archive, fond/collection and id (see #178) inside of the `atom:id` is not appropriate. Beside this, creating the dependencies (archive, etc) from the collection hierarchy is not even better.

A suggestion is to store such metadata as archive, fond, collection in constructed field to an index (see http://exist-db.org/exist/apps/doc/lucene.xml. This fields are most of the time fixed and don't need to be changed. So we dont have any trouble to get the meta infos from a charter and we dont have to deal with arrays aso...

Suggestions?

**Comment by GVogeler** [comment]
Well I'm not completely convinced that the "IT point of view" is the only valid one. Using the collection hierarchy could be a useful way to express hierarchical relationships between objects. But this would need to put metadata to the collection / archive / fond as documents into the containing collections, which is not easy to handle.
But nevertheless possible solutions could be:
1. put the information into `cei:witnessOrig/archIdentifier` with using `cei:arch/@n`, `cei:archFond/@n` for the internal keys.
2. use `cei:arch` and `cei:archFond` as subelements to `atom:entry`.
In both cases the semantics for collections is not very clear. `cei:archFond` without `ceiarch`? Or create an element from a new namespace (e.g. xrx:collection`)? Or use EAD elements?
So introduction of a dedicated vocabulary from xrx-namespace seems a way to go: `xrx:archive`, `xrx:archFond`, `xrx:collection`.
What do you think?

**Comment by GVogeler** [comment]
In any case: `atom:id` has to be unique for the whole database (and should remain as 'speaking' as possible), so we would create a duplication of information in atom:id and in separated fields, if we don't put the seperated information as sub-elemens to `atom:id`

**Comment by StephanMa** [comment]
Sounds good...

But just another example that it is necessary to build unique identifier for archive etc...

BaseX does not handle collections like eXist does. So it is not possible to build such a structure with collections. In this case, all relevant informations has to be derived from the `atom:id` which will cost a lot of processing time.

**Comment by GVogeler** [comment]
so let's consider
```
<atom:id>tag:www.monasterium.net,2011:/<xrx:type>charter</xrx:type>/<xrx:collection type="collection">IlluminierteUrkunden</xrx:collection>/<xrx:local-id>1339-11-05_Wien</xrx.local-id></atom:id>
```
or
```
<atom:id>tag:www.monasterium.net,2011:/<xrx:type>charter</xrx:type>/<xrx:collection type="archive">AT-DAW</xrx:collection>/<xrx:subcollection type="archFond">Urkunden</xrx:subcollection>/<xrx:local-id>13391105</xrx:local-id></atom:id>
```
as working hypothesis?

This we could even introduce in the current code as `//atom:id/tokenize(.,'/')` should return the same values as currently.

**Comment by StephanMa** [comment]
Does tokenize not only returns an array of values? Well maybe there is a function which returns the value of a node as text regardless of mixed content

**Comment by GVogeler** [comment]
yes, `tokenize()` returns a sequence of nodes ('array' in non-XML peoples world :-)), which is the - my impression - the usual way how mom-ca currently extracts collection / archive / fond from the atom:id. So this method should not be affected.
The xpath-way to extract only text from mixed content is `string()` - which is already heavy in use in mom-ca, as XQuery usually returns nodes, so to get to the content of a node, you need to access it via `$node/string()`.
